api/data_refinery_api/views.py

```diff
@@ -889,18 +892,32 @@ class Stats(APIView):
         enum=('day', 'week', 'month', 'year',)
     )])
     def get(self, request, format=None):
+        cached_stats = cache.get("stats", None)
         range_param = request.query_params.dict().pop('range', None)
 
+        # This should only need to run once for each range at most. From then on,
+        # the cache is refreshed in the background using apscheduler.
+        # We index the dict by str(range_param) because None is the default range.
+        if cached_stats is None or cached_stats.get(str(range_param), None) is None:
+            cached_stats = Stats.update_cache(range_param)
```

Comment:
Will the cache be invalidated if there's a new deploy?

Reply:
It looks like it will, so if someone views the dashboard within `STAT_CACHE_REFRESH_INTERVAL_MIN` minutes of deploy they won't get a cached result.